src/completions.jl

```diff
@@ -232,29 +241,36 @@ function completiondetail_keyword!(comp)
   comp[:description] = completiondescription(getdocs(Main, comp[:text]))
 end
 
+const GENSYM_REGEX = r"#\d+#\d+"
 using JuliaInterpreter: sparam_syms
 using Base.Docs
 
+# NOTE: this is really hacky, we may want to revert back to unlazy completion return type inference
 function completiondetail_method!(comp)
   mod = getmodule(comp[:rightLabel])
-  fstr, fhash = split(comp[:detailtype], ',')
-  f = getfield′(mod, Symbol(fstr))
-  isundefined(f) && return
-  # HACK: reconstruct a Method object from its hash string
+  fstr, input_types_str, fhash = split(comp[:detailtype], ' ')
+
+  # reconstruct function itself handling gensyms
+  if (m = match(GENSYM_REGEX, fstr)) !== nothing
+    # gensym pattern
+    fsym = Symbol(m.match)
+    ftyp = getfield′(mod, fsym)
+    (!isundefined(ftyp) && isdefined(ftyp, :instance)) || return
+    f = ftyp.instance
```

Comment:
Hm, so gensym handling here is no so robust, e.g. when a module prefix is also gensymed or so.
I feel it might be better to implement lazy return type inference as a future work, to make a release ?